src/client/clientmap.cpp

```diff
@@ -1077,7 +1076,7 @@ void ClientMap::renderMapShadows(video::IVideoDriver *driver,
 			++material_swaps;
 		}
 
-		v3f block_wpos = intToFloat(descriptor.m_pos / mesh_chunk_vol * mesh_chunk_vol * MAP_BLOCKSIZE, BS);
+		v3f block_wpos = intToFloat(mesh_grid.getMeshPos(descriptor.m_pos) * MAP_BLOCKSIZE, BS);
```

Comment:
Is this the same? We are dividing/multiplying with size^3, not size.

Reply:
Dividing+multiplying by size^3 is definitely not intentional, it should be size^1, and the math in getMeshPos is more correct for negative numbers.

The code originates from the very first commit in octa_block PR where I aligned all mesh generation to a grid with a larger step while still generating a mesh per mapblock. This alignment allows for easy stitching/splitting of mapblock meshes within the "octamesh" to keep mesh generation fast.

We can as well remove it now, together with the code in mapblock_mesh and mesh collector, without any impact to the visual qiality or performance.

Reply:
I see. Makes sense. Tested and works fine.